SECURITY.md

```diff
@@ -0,0 +1,79 @@
+
+## Document Phases
+These guidelines and this document are meant to evolve, the follow list captures the different phases this document will undergo.  Some areas might be complete ahead of others. Some areas or items might be marked as invalid/removed using markdown's strike-through.
+
+1. [X] Agreement of areas and 'as-is' state of OpenSearch Plugins and Extensions. **<-- Doc is here**
+2. [ ] All areas have recommendations and areas of investigation are filed as issues and linked back on this document.
+3. [ ] All investigation conclusions are captured and linked in this document, effectively define the scope of work for these areas.  Implementation of work is can be completed or outstanding.
+4. [ ] All planned work has been completed, issues around this work can be completed or outstanding. 
+5. [ ] Document complete, future work and issue will be captured out of band instead of as updates this document.
+
+## Terms
+To keep concepts consistent, this document is using terminology from [NIST Glossary](https://csrc.nist.gov/glossary).
+
+Additional terms:
+* **Plugin** - reference to the existing functionality to extend OpenSearch functionality.
+* **Extension** - reference to the in development functionality to extend OpenSearch.
+
+# Areas
+
+## Host security
+
+Plugins depend on use of the Java Security Manager is use to limit interactions on the host operation system resources (cpu/disk/memory/network/...).  JSM has been deprecated, with its removal scheduled in the next release of the JVM, see [OpenSearch discussion](https://github.com/opensearch-project/OpenSearch/issues/1687). Additional measures are needed to protect system resources.
+
+Extensions are sandboxed from the host system by operating via REST APIs.  This security boundary isolates extensions from executing operation system calls directly on OpenSearch hosts.
+
+## Communications security (COMSEC)
+
+Plugins are loaded into the same java virtual machine instance allowing communicate to OpenSearch through in process java APIs.  Plugins can issue REST API requests to the OpenSearch hosts reusing the standard node-to-node communications, internally called the transport client.
+
+Extensions of OpenSearch communicate via https requests between the nodes on the cluster and the extensions endpoint(s).  This is a bi-direction communication also allows extensions to contact the OpenSearch cluster through its available REST APIs.
+
+## Data Security
+
+OpenSearch stores data in memory and local file system storage.  This data is stored unencrypted.
+
+Plugins can use the existing data systems of the OpenSearch.  Several classes of plugins extend storage options out to external services.
+
+### Access Control
+
+With the security plugin installed, role based access control (RBAC) is available with a proprietary policy document format.  Access control over native OpenSearch data is possible with this plugin installed.
```

Comment:
I think the word proprietary might not be the one you're looking for here (see https://en.wikipedia.org/wiki/Proprietary_file_format). Perhaps "specific to OpenSearch", "non-standard" (if there is a standard for these...?